wit/store.wit

```diff
@@ -7,22 +7,62 @@
 /// ensuring compatibility between different key-value stores. Note: the clients will be expecting
 /// serialization/deserialization overhead to be handled by the key-value store. The value could be
 /// a serialized object from JSON, HTML or vendor-specific data types like AWS S3 objects.
-/// 
-/// Data consistency in a key value store refers to the guarantee that once a write operation
-/// completes, all subsequent read operations will return the value that was written.
+///
+/// ## Consistency
 /// 
 /// Any implementation of this interface must have enough consistency to guarantee "reading your
-/// writes." In particular, this means that the client should never get a value that is older than
-/// the one it wrote, but it MAY get a newer value if one was written around the same time. These
-/// guarantees only apply to the same client (which will likely be provided by the host or an
-/// external capability of some kind). In this context a "client" is referring to the caller or
-/// guest that is consuming this interface. Once a write request is committed by a specific client,
-/// all subsequent read requests by the same client will reflect that write or any subsequent
-/// writes. Another client running in a different context may or may not immediately see the result
-/// due to the replication lag. As an example of all of this, if a value at a given key is A, and
-/// the client writes B, then immediately reads, it should get B. If something else writes C in
-/// quick succession, then the client may get C. However, a client running in a separate context may
-/// still see A or B
+/// writes." In particular, this means that a `get` call for a given key on a given `bucket`
+/// resource should never return a value that is older than the the last value written to that key
+/// on the same resource, but it MAY get a newer value if one was written around the same
+/// time. These guarantees only apply to reads and writes on the same resource; they do not hold
+/// across multiple resources -- even when those resources were opened using the same string
+/// identifier by the same component instance.
+///
+/// The following pseudocode example illustrates this behavior.  Note that we assume there is
+/// initially no value set for any key and that no other writes are happening beyond what is shown
+/// in the example.
+///
+/// bucketA = open("foo")
+/// bucketB = open("foo")
+/// bucketA.set("bar", "a")
+/// // The following are guaranteed to succeed:
+/// assert bucketA.get("bar").equals("a")
+/// assert bucketB.get("bar").equals("a") or bucketB.get("bar") is None
+/// // ...whereas this is NOT guaranteed to succeed immediately (but should eventually):
+/// // assert bucketB.get("bar").equals("a")
```

Comment:
It sounds like from what's written above this is not guranteed to ever be true. Since consistency is not guaranteed across resource handles, `bucketB.get("bar")` may never equal `"a"` even with unlimited time and no other writes.

Reply:
Right, hence the "should".  I think it's worth mentioning what a conforming implementation should make a best effort to do (i.e. in normal operation, barring exceptional circumstances) as well as what it _must_ do.

Reply:
If we're using the RFC 2119 meaning of "should" I think we should write it as "SHOULD" (in all caps). A non-RFC definition of "should" here might lead readers to interpret "should" as "will".